uberfire-extensions/uberfire-wires/uberfire-wires-core/uberfire-wires-core-grids/src/main/java/org/uberfire/ext/wires/core/grids/client/widget/grid/selections/impl/BaseCellSelectionManager.java

```diff
@@ -254,6 +337,22 @@ private boolean moveSelection(final GridData.SelectedCell origin,
                           false);
     }
 
+    private boolean moveFromDataToHeader(final int uiColumnIndex) {
+        final int uiHeaderRowIndex = ColumnIndexUtilities.getMaxUiHeaderRowIndexOfColumn(gridModel, uiColumnIndex);
+        return selectHeaderCell(uiHeaderRowIndex,
+                                uiColumnIndex,
+                                false,
+                                false);
+    }
+
+    private boolean possibleToSelectHeaderCell() {
+        return !simpleInnerGridIsSelected() && gridModel.getHeaderRowCount() > 0;
+    }
+
+    private boolean simpleInnerGridIsSelected() {
```

Comment:
I think this is DMN specific? What purpose does it serve? Does it relate to `LiteralExpressionGrid` and `UndefinedExpressionGrid`?

Reply:
Yes, related to those, should I remove it according to you feedback on kie-wb-common of the ensemble?

Reply:
@jomarko I'd like to understand _what_ this does? It suggests you can **only** move to the header _if_ the current grid is not a `LiteralExpressionGrid` or `UndefinedExpressionGrid`. What about an "top-level `LiteralExpressionGrid`" that has one row, one columns **and** a header cell? I'll try this scenario in a minute.. when I retest the `kie-wb-common` PR.

Reply:
@manstis yes, you are right **What about an "top-level LiteralExpressionGrid" that has one row, one columns and a header cell?** doesn't work, so will check after you open a PR for not adding header into nested grids.